bcgov/HOUS-permit-portal
# Feature/bphh 1540/configure reasons

## Description

Allow the super admin to create custom reasons for revision under configuration management

- only 20 reasons are allowed (10 may be too restrictive)
- revisions can be deleted (archived) but are still referenced by existing revision requests
- only 200 reason codes can exist in total (archived and un-archived) as all of these are loaded in the site configuration
- revision requests reference reasons by reason_code, this is the identity of the reason
- reasons re-created under a previously archived reason code is equivalent to un-archiving and updating the reason
- reasons description can be edited and the new text will appear wherever that reason code is referenced

## What type of PR is this? (check all applicable)

- [ x] 🍕 Feature

## Related Tickets & Documents

https://hous-bssb.atlassian.net/browse/BPHH-1540

## Steps to QA

See card. You might also want to check that reason text changes when looking at a revision requested permit application.
